Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -391,33 +391,19 @@ NiftiImageIO::GenerateStreamableReadRegionFromRequestedRegion(const ImageIORegio
 }
 
 
-// This internal proxy class provides a pointer-like interface to a nifti_image*, by supporting
-// conversions between proxy and nifti_image pointer and arrow syntax (e.g., m_NiftiImage->data).
-class NiftiImageIO::NiftiImageProxy
+// Custom deleter for nifti_image_struct
+void
+NiftiImageIO::NiftiImageDeleter::operator()(nifti_image_struct * ptr) const
 {
-  nifti_image * m_ptr;
-
-public:
-  NiftiImageProxy(nifti_image * ptr)
-    : m_ptr(ptr)
-  {}
-
-  operator nifti_image *() { return m_ptr; }
-
-  nifti_image *
-  operator->()
+  if (ptr)
```

Comment:
No need for this NULL check, like free(NULL), nifti_image_free(NULL) does nothing.